File: cci_1-4.py
"""
Cracking the coding interview - Exercise 1.4
March 4th, 2017
"""
import sys


def replace_space(testString):
    """
    This function replaces any spaces present in a provided string
    with the char sequence "%20"
    """
    newString = ""

    for c in testString:
        if c != ' ':
            newString += c
        elif c == ' ':
            newString += "%20"

    print(newString)
    sys.exit()


testString = input("\nPlease enter a string to be modifed: ")

replace_space(testString)

File: cci_1-4.cc
//================================================
// Cracking the coding interview - Exercise 1.4
// March 3rd, 2017
//================================================

#include <iostream>
#include <string>
#include <stdlib.h>
#include <queue>

std::string replace_space(std::string testStr)
{// This function replaces spaces present in a string with "%20"
    std::queue<char> newString; // queue used for fun here - not optimal
    std::string modiString;

    for (auto c: testStr) {
        if (c != ' ') {
            newString.push(c);
        }
        else if (c == ' ') {
            newString.push('%');
            newString.push('2');
            newString.push('0');
        }
    }

    while (!newString.empty()) {
        modiString += newString.front();
        newString.pop();
    }

    return modiString;
}

int main()
{
    std::string testString, modifiedString;

    std::cout << "\nPlease enter a string to be modified: \n";
    std::getline (std::cin, testString);

    modifiedString = replace_space(testString);

    std::cout << "\n" << modifiedString << "\n";

    return 0;
}
